addons/mail/models/res_partner.py

```diff
@@ -35,7 +36,19 @@ def _compute_contact_address_inline(self):
             partner.contact_address_inline = re.sub(r'\n(\s|\n)*', ', ', partner.contact_address).strip().strip(',')
 
     def _compute_im_status(self):
-        super()._compute_im_status()
+        status_by_partner = {}
+        for presence in self.env["mail.presence"].search([("user_id", "in", self.user_ids.ids)]):
```

Comment:
Also here, if we have the inverse field on user, we can potentially save a query with cache (if we call status of both user and partner in the same request for example).